censys/asm/events.py

```diff
@@ -0,0 +1,98 @@
+"""
+Class for interfacing with the Censys Logbook API.
+"""
+from typing import Optional, Generator, List
+from datetime import datetime
+
+
+class Events:
+    """
+    Events API class
+    """
+
+    def __init__(self, client):
+        self.client = client
+        self.base_path = "logbook"
+
+    def get_cursor(
+        self, start: Optional[datetime] = None, filters: Optional[List[str]] = None
+    ) -> str:
+        """
+        Requests a logbook cursor.
+
+        Args:
+            start (datetime, optional): Seed type ['IP_ADDRESS', 'DOMAIN_NAME', 'CIDR', 'ASN'].
+            filters (list, optional): List of filters applied to logbook search results.
+
+        Returns:
+            str: Cursor result.
+        """
+
+        path = f"{self.base_path}-cursor"
+        data = format_data(start=start, filters=filters)
+
+        return self.client._post(path, data=data)["cursor"]
```

Comment:
This line and a lot of the other functions are trying to use the protected member _get of a client class `(protected-access)`. To solve this we can create a class that has the BASE_URL of ASM rather than Search, this allows us to use protected members.

Reply:
This helps explain the comment above.